tests/rptest/tests/partition_move_interruption_test.py

```diff
@@ -0,0 +1,270 @@
+    """
+    def __init__(self, ctx, *args, **kwargs):
+        super(PartitionMoveInterruption, self).__init__(
+            ctx,
+            *args,
+            extra_rp_conf={
+                # Disable leader balancer, as this test is doing its own
+                # partition movement and the balancer would interfere
+                'enable_leader_balancer': False
+            },
+            **kwargs)
+        self._ctx = ctx
+        self.throughput = 10000
+        self.moves = 10
+        self.min_records = 100000
+        self.partition_count = 20
+
+    def replace_replicas(self, admin, assignments):
+        selected, replacements = self._choose_replacement(admin,
+                                                          assignments,
+                                                          allow_no_ops=False)
+        selected = sorted(selected, key=lambda v: v['node_id'])
+        replacements = sorted(replacements, key=lambda v: v['node_id'])
+        while selected == replacements:
+            selected, replacements = self._choose_replacement(
+                admin, assignments, allow_no_ops=False)
+
+        return selected, replacements
+
+    def _random_move(self, unclean_abort):
+        admin = Admin(self.redpanda)
+        partition = random.randint(0, self.partition_count - 1)
+
+        assignments = self._get_assignments(admin, self.topic, partition)
+        prev_assignments = assignments.copy()
+
+        self.logger.info(
+            f"assignments for {self.topic}-{partition}: {prev_assignments}")
+
+        # build new replica set by replacing a random assignment, do not allow no ops as we want to have operation to cancel
+        selected, replacements = self.replace_replicas(admin, assignments)
+
+        self.logger.info(
+            f"replacement for {self.topic}-{partition}:{len(selected)}: {selected} -> {replacements}"
+        )
+        self.logger.info(
+            f"new assignments for {self.topic}-{partition}: {assignments}")
+
+        admin.set_partition_replicas(self.topic, partition, assignments)
+
+        def move_in_progress():
+            p = admin.get_partitions(self.topic, partition)
+            return p['status'] == 'in_progress'
+
+        wait_until(move_in_progress, 10, 1)
+
+        try:
+            if unclean_abort:
+                admin.force_abort_partition_move(self.topic, partition)
+            else:
+                admin.cancel_partition_move(self.topic, partition)
+        except requests.exceptions.HTTPError as e:
+            # we do not throttle cross core moves, it may already be finished
+            assert e.response.status_code == 400
+            return
+
+        # wait for previous assignment
+        self._wait_post_move(self.topic, partition, prev_assignments, 60)
+
+    def _throttle_recovery(self, new_value):
+        rpk = RpkTool(self.redpanda)
+        rpk.cluster_config_set("raft_learner_recovery_rate", str(new_value))
```

Comment:
There's a `set_cluster_config` helper on RedpandaService that you might like to use instead of RPK